react/components/ReviewBlock.tsx

```diff
@@ -56,12 +56,12 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
     return Promise.all(
       data.map(async (item: any) => {
         const res: any = await checkRestriction(item.refid)
-        if (
-          res?.data?.productSuggestions?.products[0]?.items[0]?.itemId ===
-          item.sku
-        ) {
-          return item
-        }
+
+        const foundSku = res?.data?.productSuggestions?.products[0]?.items.find(
+          (suggestedItem) => suggestedItem.itemId === item.sku
+        )
+
+        return foundSku ? item : null
       })
     )
   }
```

Comment:
Have you meet the case that the matched refid showed at the later index? This change can increase the complexity to square growth.

Reply:
Yes, the SKU ID for "880020a" is "880020" which is the second item in the list when running the `productSuggestions` query. I only stumbled upon one example so I'm not sure if this would be a common occurrence.